plugins/graphite/check-data.rb

```diff
@@ -179,6 +179,14 @@ def retrieve_data
           url_opts[:http_basic_authentication] = [config[:username], pass.chomp]
         end # we don't have both username and password trying without
 
+        no_proxy = ENV["no_proxy"]
+        unless no_proxy.nil?
+          domain_list = no_proxy.split(',')
+          if domain_list.any? { |domain| config[:server].include?(domain) }
+            url_opts[:proxy] = config[:proxy]
```

Comment:
Where is this `config[:proxy]` coming from ?
check-data.rb has no such parameter.

Reply:
Nowhere, its to pass the proxy as nil.
But you are right, i better just set url_opts[:proxy] to nil to make it clear.